jdaviz/configs/default/plugins/subset_tools/subset_tools.py

```diff
@@ -839,6 +848,42 @@ def _set_value_in_subset_definition(self, index, name, desired_key, new_value):
                 self.subset_definitions[index][i]['value'] = new_value
                 break
 
+    @cached_property
+    def selected_subset_group(self):
+        for subset_group in self.app.data_collection.subset_groups:
+            if subset_group.label == self.subset.selected:
+                return subset_group
+
+    def rename_subset(self, old_label, new_label):
+        """
+        Method to rename an existing subset
+
+        Parameters
+        ----------
+        old_label : str
+            The current label of the subset to be renamed.
+        new_label : str
+            The new label to apply to the selected subset.
+        """
+        self.app._rename_subset(old_label, new_label)
+        self._sync_available_from_state()
+
+    def rename_selected(self, new_label):
```

Comment:
Is this more performant or just a shortcut (in which case I'm not sure if its worth the extra code and test-coverage or not, but don't have a strong opinion)?

Just a note to future selves: eventually we likely will want to migrate this logic into the subset dropdown itself (so these would become `subset_tools.subset.rename...`) since we're planning to use an editable select component, but I think it still makes sense to expose methods at the top-level in this plugin for convenience, so we can always just move the logic there and then have these methods call that instead.